core/src/main/scala/com/mesosphere/usi/core/Scheduler.scala

```diff
@@ -65,18 +61,16 @@ import scala.util.{Failure, Success}
   * }}}
   */
 object Scheduler {
-  type SpecInput = (SpecsSnapshot, Source[SpecUpdated, Any])
-
-  type StateOutput = (StateSnapshot, Source[StateEvent, Any])
+  type StateOutput = (StateSnapshot, Source[StateEventOrSnapshot, Any])
```

Comment:
```suggestion
  type StateOutput = (StateSnapshot, Source[StateEvent, Any])
```

or simply `type StateOuput = StateEventOrSnapshot`.

Reply:
Yeah, lets do this.